storage_controller/src/http.rs

```diff
@@ -1380,6 +1383,30 @@ async fn handle_status(req: Request<Body>) -> Result<Response<Body>, ApiError> {
     json_response(StatusCode::OK, ())
 }
 
+/// Liveness endpoint indicates that this storage controller is in a state
+/// where it can fulfill it's responsibilties. Namely, startup has finished
```

Comment:
_or_ it's in a position to forward requests to someone else -- when we're no longer the leader but forwarding requests to the new leader, we're going to return 200 here, right?

Reply:
No, we will return 503 in that case. It's an interesting point though. Would k8s restart the pod from the old replica set if there's already a new live pod from the new replica set (the one we are forwarding to)? I don't know the answer but the liveness probe configuration should paper over this concern (needs to fail for a number of times to be considered not live - by then the old replica set should scale down anyway).

I wouldn't be opposed to extending liveness to say successfully forwarding pods are allowed, but it's tricky to differentiate between a pod that's forwarding everything fine and a pod that ended up in stepped down state without another leader like the example in the cover letter.